port/src/allocator.rs

```diff
@@ -489,17 +488,23 @@ mod global {
     /// A GlobalHeap is an aligned wrapper around an owned
     /// buffer.
     #[repr(C, align(4096))]
-    struct GlobalHeap([u8; GLOBAL_HEAP_SIZE]);
+    pub struct GlobalHeap([u8; GLOBAL_HEAP_SIZE]);
     impl GlobalHeap {
-        const fn new() -> GlobalHeap {
+        pub const fn new() -> GlobalHeap {
             Self([0u8; GLOBAL_HEAP_SIZE])
         }
     }
 
+    impl Default for GlobalHeap {
```

Comment:
Hmm, do we really want to `Default` this?

Reply:
It was added at the suggestion of clippy.  If we want to keep the code clippy-happy, we could do this or disable this: https://rust-lang.github.io/rust-clippy/master/index.html#new_without_default

Reply:
Hmm. I don't suppose it hurts anything, but it is kind of annoying. :-/

Reply:
Yeah, I don't really see the point in this case either - that's the problem with these tools becoming too opinionated while we want to be clean.